packages/netstring/test/test-netstring.js

```diff
@@ -20,81 +20,115 @@ async function read(source) {
   return array;
 }
 
-test('read short messages', async t => {
-  const r = makeNetstringReader([encoder.encode('0:,1:A,')], {
-    name: '<unknown>',
-    capacity: 1,
-  });
-  const array = await read(r);
-  t.deepEqual(
-    ['', 'A'],
-    array.map(chunk => decoder.decode(chunk)),
-  );
-});
-
-test('read a message divided over a chunk boundary', async t => {
+const readChunkedMessage = async (t, chunkStrings, expectedDataStrings) => {
   const r = makeNetstringReader(
-    [encoder.encode('5:hel'), encoder.encode('lo,')],
+    chunkStrings.map(chunkString => encoder.encode(chunkString)),
     {
       name: '<unknown>',
-      capacity: 1,
     },
   );
   const array = await read(r);
   t.deepEqual(
-    ['hello'],
+    expectedDataStrings,
     array.map(chunk => decoder.decode(chunk)),
   );
-});
+};
 
-test('read messages divided over chunk boundaries', async t => {
+test('read short messages', readChunkedMessage, ['0:,1:A,'], ['', 'A']);
+test(
+  'read short messages with data divided over chunk boundaries',
+  readChunkedMessage,
+  ['0:', ',', '1:A', ','],
+  ['', 'A'],
+);
+
+test(
+  'read a message in single chunk',
+  readChunkedMessage,
+  ['5:hello,'],
+  ['hello'],
+);
+test(
+  'read a message with data in separate chunk',
+  readChunkedMessage,
+  ['5:', 'hello,'],
+  ['hello'],
+);
+test(
+  'read a message with data divided over a chunk boundary',
+  readChunkedMessage,
+  ['5:hel', 'lo,'],
+  ['hello'],
+);
+
+test(
+  'read messages divided over chunk boundaries',
+  readChunkedMessage,
+  ['5:hello', ',5:world,8:good ', 'bye,'],
+  ['hello', 'world', 'good bye'],
+);
+
+test(
+  'read prefix colon divided over chunk boundary',
+  readChunkedMessage,
+  ['0', ':,', '1', ':A,'],
+  ['', 'A'],
+);
+
+test(
+  'read length prefix divided over chunk boundaries',
+  readChunkedMessage,
+  ['1', '1:hello world,'],
+  ['hello world'],
+);
+
+const readErroneousChunkedMessage = async (t, chunkStrings) => {
   const r = makeNetstringReader(
-    [
-      encoder.encode('5:hel'),
-      encoder.encode('lo,5:world,8:good '),
-      encoder.encode('bye,'),
-    ],
+    chunkStrings.map(chunkString => encoder.encode(chunkString)),
     {
       name: '<unknown>',
-      capacity: 1,
     },
   );
-  const array = await read(r);
-  t.deepEqual(
-    ['hello', 'world', 'good bye'],
-    array.map(chunk => decoder.decode(chunk)),
-  );
-});
+  return t.throwsAsync(() => read(r));
+};
+
+test('fails reading invalid prefix', readErroneousChunkedMessage, ['1.0:A,']);
+test('fails reading incomplete data', readErroneousChunkedMessage, ['5:hello']);
+test('fails reading invalid separator', readErroneousChunkedMessage, ['0:~']);
+test('fails reading no colon', readErroneousChunkedMessage, ['1A,']);
 
 function delay(ms) {
   return new Promise(resolve => {
     setTimeout(resolve, ms);
   });
 }
 
-const makeArrayWriter = () => {
+const makeArrayWriter = opts => {
```

Comment:
Very ignorable nit: does our style guide have anything to say about abbreviations? My personal preference is to spell these out.

Reply:
Arguably I also prefer spelling out but used the term `opts` because I copied it from the JSDoc `@param` of `makeNetstringWriter`, which I itself copied from the existing reader one.

Reply:
_mea culpa_!